crates/control/src/role_assignment.rs

```diff
@@ -254,6 +257,9 @@ impl RoleAssignment {
                                     robot_to_field,
                                     cycle_start_time,
                                 ),
```

Comment:
If i see this correctly here, only the ball_position is different in the if-else branch. If that's the case I'd prefer to only compute the ball_position in the if-else and do the network message after that.

Reply:
Yes that's true, I've rearranged it like suggested

Reply:
Looks much better, thanks